src/java/org/apache/cassandra/streaming/async/StreamCompressionSerializer.java

```diff
@@ -94,15 +87,15 @@ public ByteBuffer deserialize(LZ4FastDecompressor decompressor, DataInputPlus in
 
         if (in instanceof ReadableByteChannel)
         {
-            ByteBuffer compressed = BufferPool.get(compressedLength);
+            ByteBuffer compressed = ByteBuffer.allocateDirect(compressedLength);
             ByteBuffer uncompressed = null;
             try
             {
                 int readLength = ((ReadableByteChannel) in).read(compressed);
                 assert readLength == compressed.position();
                 compressed.flip();
 
-                uncompressed = BufferPool.get(uncompressedLength);
+                uncompressed = ByteBuffer.allocateDirect(uncompressedLength);
```

Comment:
Good call. I've switched to use the `ByteBuffer`s I can get from a `ByteBuf`. Further, because of the way that netty give you the nio buffer, i don't run into the 'arrayOffset' bug in lz4 - which means I can ditch having an intermediate buffer to compress into/copy from :).

I also converted the API of this class, and fixed up the downstream callers.